src/SME.SGP.Aplicacao/CasosDeUso/FechamentoAcompanhamentoTurmas/ExecutarConsolidacaoTurmaConselhoClasseAlunoUseCase.cs

```diff
@@ -108,23 +129,37 @@ public async Task<bool> Executar(MensagemRabbit mensagemRabbit)
 
                 var consolidadoTurmaAlunoId = await repositorioConselhoClasseConsolidado.SalvarAsync(consolidadoTurmaAluno);
 
-                var consolidadoNota = await repositorioConselhoClasseConsolidadoNota.ObterConselhoClasseConsolidadoPorTurmaBimestreAlunoNotaAsync(consolidadoTurmaAlunoId, filtro.Bimestre, filtro.ComponenteCurricularId);
-                if (consolidadoNota == null) 
-                    consolidadoNota = new ConselhoClasseConsolidadoTurmaAlunoNota() 
-                    { 
-                        ConselhoClasseConsolidadoTurmaAlunoId = consolidadoTurmaAlunoId,
-                        Bimestre = filtro.Bimestre,
-                    };
+                double? nota = null;
+                double? conceito = null;
+                if (conselhoClasseNotasAluno != null && conselhoClasseNotasAluno.Any())
+                {
+                    nota = conselhoClasseNotasAluno.First().Nota;
```

Comment:
Esses pontos considerando o primeiro item da lista podem dar erro de referência nula.
Acredito que é necessário usar o "FirstOrDefault()" e testar se o objeto não está nulo.
E se for o caso disparar uma exceção de negócio.